docs/migrate-with-crucible.md

```diff
@@ -0,0 +1,172 @@
+# Running a live migration "by hand" with a crucible boot disk
+
+In the product, live migration is managed by nexus. Still, it is extremely
+useful for development to be able to test software components in isolation. One
+obstacle for testing inter-machine migration in propolis without a full control
+plane is the need for shared storage — in particular, a source of shared storage
+for the guest's boot disk.
+
+Since crucible will be providing storage in the product, I chose to get
+that working over other options. This document has some instructions on how to
+get propolis to use crucible as a backend for a boot disk. At the moment it's
+not the most user-friendly experience, but since it took some effort to figure
+out, I wanted to at least capture what I did.
+
+## Requirements
+
+For this setup, you'll need:
+- a "source" propolis server
+- a "destination" propolis server (on the same machine or otherwise)
+- a propolis CLI
+- a copy of [crucible](https://github.com/oxidecomputer/crucible) and a place to
+  run downstairs processes that the source/destination machines can access
+- an OS image that you'd like to boot
+
+## Setup
+
+### Seed crucible downstairs with the OS image
+
+From the machine where you will run the crucible downstairs, set up 3 crucible
+downstairs regions using the `--import-path` flag to specify where the OS image
+is. (See the crucible documentation for information about other flags.)
+
+For example: 
+```
+$ ./target/release/crucible-downstairs create --import-path /home/jordan/images/helios-generic-ttya-base_20230109.raw --data region8810 --uuid $(uuidgen) --extent-size 64000 --extent-count 64
+
+$ ./target/release/crucible-downstairs create --import-path /home/jordan/images/helios-generic-ttya-base_20230109.raw --data region8820 --uuid $(uuidgen) --extent-size 64000 --extent-count 64
+
+$ ./target/release/crucible-downstairs create --import-path /home/jordan/images/helios-generic-ttya-base_20230109.raw --data region8830 --uuid $(uuidgen) --extent-size 64000 --extent-count 64
+```
+
+Each `create` will setup a region file. In the above example, these files are
+`region8810`, `region8820`, and `region8830`, respectively.
+
+### Run the crucible downstairs
+
+After seeding the downstairs with the image, run the downstairs processes.
+
+For example:
+```
+$ ./target/release/crucible-downstairs run -d region8810 -p 8810 -a 172.20.3.73
+$ ./target/release/crucible-downstairs run -d region8820 -p 8820 -a 172.20.3.73
+$ ./target/release/crucible-downstairs run -d region8830 -p 8830 -a 172.20.3.73
```

Comment:
You might want to mention that the `-p` is the port number this downstairs will use, and the `-a` is the I.P. address where the downstairs will listen.  It can be localhost or the external I.P. address of the machine where you are running the downstairs.  Also that these IP:Port will be used below in the JSON file.